### PR TITLE
Remove comments from old PropTypes

### PR DESCRIPTION
**Pull request by Inbal-Tish**
1) Remove comments from old propTypes for cleaner code.
2) Use 'sameMonth' instead of string compare.
3) Use 'getDefaultLocal'.
4) Move 'isPastDate' to dateutils.
5) Optional params in 'sameDate', 'sameMonth' and 'parseDate'.

### CODE REVIEW

File: src/calendar/header/index.tsx [hunk 1]
```diff
@@ -30,12 +30,17 @@ import styleConstructor from './style';
 import {Theme, Direction} from '../../types';
 
 export interface CalendarHeaderProps {
+  month?: XDate;
```
Comment (lidord-wix) [inline src/calendar/header/index.tsx:33]: Should be replaced by string as well?

File: src/calendar/index.tsx [hunk 2]
```diff
@@ -26,8 +26,6 @@ type MarkedDatesType = {
 };
 
 export interface CalendarProps extends CalendarHeaderProps, DayProps {
-  /** Specify theme properties to override specific styles for calendar parts */
-  theme?: Theme;
```
Comment (lidord-wix) [inline src/calendar/index.tsx:30]: Is it safe to remove all these props?

**Review by lidord-wix — COMMENTED** [review]
Looks good!
I wrote few comments to make sure we don't break anything :)

File: src/agenda/index.tsx [hunk 3]
```diff
@@ -32,10 +31,6 @@ const HEADER_HEIGHT = 104;
 const KNOB_HEIGHT = 24;
 
 export type AgendaProps = CalendarListProps & ReservationListProps & {
-  /** the list of items that have to be displayed in agenda. If you want to render item as empty date
-   the value of date key has to be an empty array []. If there exists no value for date key it is
-   considered that the date in question is not yet loaded */
-  items?: AgendaSchedule;
```
[the remark this answers is not in the document]
Comment (Inbal-Tish) [reply src/agenda/index.tsx:38]: Agenda get this prop from `ReservationListProps`. I don't see the need for the duplication

File: src/calendar/header/index.tsx (hunk 1 above)
Comment (Inbal-Tish) [reply src/calendar/header/index.tsx:33]: It should be a private prop... We'll need to sort all these props issues in V2, it's currently a big mess

File: src/calendar/index.tsx (hunk 2 above)
Comment (Inbal-Tish) [reply src/calendar/index.tsx:30]: They are header props, but I'll put them back and we'll sort these props issues in another occasion

Comment (Inbal-Tish) [reply src/calendar/index.tsx:30]: Done

File: src/agenda/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Inbal-Tish) [reply src/agenda/index.tsx:38]: Done

**Review by lidord-wix — APPROVED** [review]

**Pull request merged**